cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/NamespaceHttpHandler.java

```diff
@@ -179,6 +184,39 @@ public void delete(HttpRequest request, HttpResponder responder, @PathParam("nam
       responder.sendStatus(HttpResponseStatus.OK);
     } catch (NotFoundException e) {
       responder.sendString(HttpResponseStatus.NOT_FOUND, String.format("Namespace %s not found.", namespace));
+    } catch (NamespaceCannotBeDeletedException e) {
+      responder.sendString(HttpResponseStatus.FORBIDDEN, e.getMessage());
+    } catch (Exception e) {
+      LOG.error("Internal error while deleting namespace.", e);
+      responder.sendString(HttpResponseStatus.INTERNAL_SERVER_ERROR, e.getMessage());
+    }
+  }
+
+  /**
+   * DO NOT DOCUMENT THIS API
+   */
+  @DELETE
+  @Path("/unrecoverable/namespaces/{namespace-id}/datasets")
+  public void deleteDatasets(HttpRequest request, HttpResponder responder,
+                             @PathParam("namespace-id") String namespace) {
+    if (!cConf.getBoolean(Constants.Dangerous.UNRECOVERABLE_RESET, Constants.Dangerous.DEFAULT_UNRECOVERABLE_RESET)) {
+      responder.sendStatus(HttpResponseStatus.FORBIDDEN);
+      return;
+    }
+    if (isReserved(namespace)) {
```

Comment:
Updated to remove this check from here. Contents of default namespace can be deleted now.